kernel/src/transaction.rs

```diff
@@ -0,0 +1,124 @@
+use std::sync::Arc;
+
+use crate::actions::visitors::TransactionVisitor;
+use crate::actions::{get_log_schema, TRANSACTION_NAME};
+use crate::snapshot::Snapshot;
+use crate::EngineInterface;
+use crate::{actions::Transaction, DeltaResult};
+
+pub use crate::actions::visitors::TransactionMap;
+pub struct TransactionScanner {
+    snapshot: Arc<Snapshot>,
+}
+
+impl TransactionScanner {
+    pub fn new(snapshot: Arc<Snapshot>) -> Self {
+        TransactionScanner { snapshot }
+    }
```

Comment:
aside: @nicklan If I counted correctly, this brings us to six structs whose `new` is just a thin wrapper for convenience. Do you think it's worth bringing in the [derive-new](https://crates.io/crates/derive-new) crate? Tho this [reddit thread](https://www.reddit.com/r/rust/comments/18eln31/why_does_the_core_not_provide_a_new_derive/) has some arguments for why trivial new might be an antipattern?

Reply:
Mmm, it's an interesting point. I think the only reason I'd want to keep it this way is that we could easily change things in the new method and/or add new fields with default values (which we could do using `derive-new`) and not require upstream changes.

I created #181 to remind us to think about this